website/content/docs/troubleshoot/troubleshoot-services.mdx

```diff
@@ -0,0 +1,136 @@
+
+## Introduction
+
+When communication between upstream and downstream services in a service mesh fails, you can diagnose the cause manually with one or more of Consul’s built-in features, including [health check queries](/consul/docs/discovery/checks), [the UI topology view](/consul/docs/connect/observability/ui-visualization), and [agent telemetry metrics](/consul/docs/agent/telemetry#metrics-reference).
+
+The `consul troubleshoot` command performs several checks in sequence that enable you to discover issues that impede service-to-service communication. The process systematically queries the [Envoy administration interface API](https://www.envoyproxy.io/docs/envoy/latest/operations/admin) and the Consul API to determine the cause of the communication failure.
+
+The troubleshooting command validates service-to-service communication by checking for the following common issues:
+
+- Upstream service does not exist
+- One or both hosts are unhealthy
+- A filter affects the upstream service
+- The CA has expired mTLS certificates
+- The services have expired mTLS certificates
+
+Consul outputs the results of these validation checks to the terminal along with suggested actions to resolve the service communication failure. When it detects rejected configurations or connection failures, Consul also outputs Envoy metrics for services.
+
+### Envoy proxies in a service mesh
+
+Consul validates communication in a service mesh by checking the Envoy proxies that are deployed as sidecars for the upstream and downstream services. As a result, troubleshooting requires that [Consul’s service mesh features are enabled](/consul/docs/connect/configuration).
+
+For more information about using Envoy proxies with Consul, refer to [Envoy proxy configuration for service mesh](/consul/docs/connect/proxies/envoy).
+
+## Requirements
+
+- Consul v1.15 or later.
+- For Kubernetes, the `consul-k8s` CLI must be installed.
+
+### Technical constraints
+
+When troubleshooting service-to-service communication issues, be aware of the following constraints:
+
+- The troubleshooting tool does not check service intentions. For more information about intentions, including precedence and match order, refer to [service mesh intentions](/consul/docs/connect/intentions).
+- The troubleshooting tool validates one direct connection between a downstream service and an upstream service. You must run the `consul troubleshoot` command with the Envoy ID for an individual upstream service. It does support validating multiple connections simultaneously.
+- Because it validates direct communication between two Envoy proxies, the troubleshooting tool does not support checks for service-to-service connections when communication between the services passes through a mesh gateway or a terminating gateway.
```

Comment:
This is not strictly true, we actually can troubleshoot services across peered clusters (and a mesh gateway in between). I will update the RFC to reflect this. 
@erichaberkorn could you please help rewording of this section.